File: app/src/main/java/com/vtrack/data/model/MaintenanceType.kt
package com.vtrack.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "maintenance_types",
    foreignKeys = [
        ForeignKey(
            entity = Vehicle::class,
            parentColumns = ["id"],
            childColumns = ["vehicleId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index("vehicleId")
    ]
)
data class MaintenanceType(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val vehicleId: Long,
    val name: String,
    val intervalMiles: Int,
    val intervalMonths: Int? = null,
    val description: String? = null,
    val isActive: Boolean = true,
    val createdAt: Long = System.currentTimeMillis()
)

File: vtrack-ios/VTrack/Models/MaintenanceType.swift
import Foundation
import SwiftData

@Model
final class MaintenanceType {
    var vehicle: Vehicle?
    var name: String
    var intervalMiles: Int
    var intervalMonths: Int?
    var descriptionText: String?
    var isActive: Bool
    var createdAt: Date

    @Relationship(deleteRule: .cascade, inverse: \MaintenanceRecord.maintenanceType)
    var records: [MaintenanceRecord]

    init(vehicle: Vehicle, name: String, intervalMiles: Int, intervalMonths: Int? = nil, descriptionText: String? = nil) {
        self.vehicle = vehicle
        self.name = name
        self.intervalMiles = intervalMiles
        self.intervalMonths = intervalMonths
        self.descriptionText = descriptionText
        self.isActive = true
        self.createdAt = Date()
        self.records = []
    }
}
